client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -94,24 +95,43 @@ public void createReaderGroup(String groupName, ReaderGroupConfig config) {
         StateSynchronizer<ReaderGroupState> synchronizer = clientFactory.createStateSynchronizer(NameUtils.getStreamForReaderGroup(groupName),
                                               new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         Map<SegmentWithRange, Long> segments = ReaderGroupImpl.getSegmentsForStreams(controller, config);
+
         synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        @Cleanup
+        ReaderGroupImpl groupImpl = (ReaderGroupImpl) getReaderGroup(groupName);
+        val state = groupImpl.getState();
+        if (state.getConfigState() == ReaderGroupState.ConfigState.INITIALIZING && state.getConfig().equals(config)) {
+            groupImpl.doInit(state);
+        }
     }
 
     @Override
     public void deleteReaderGroup(String groupName) {
-        getAndHandleExceptions(controller.sealStream(scope, getStreamForReaderGroup(groupName))
-                                         .thenCompose(b -> controller.deleteStream(scope,
-                                                                                   getStreamForReaderGroup(groupName)))
-                                         .exceptionally(e -> {
-                                             if (e instanceof InvalidStreamException) {
-                                                 return null;
-                                             } else {
-                                                 log.warn("Failed to delete stream", e);
-                                             }
-                                             throw Exceptions.sneakyThrow(e);
-                                         }),
-                               RuntimeException::new);
-        
+        @Cleanup
+        ReaderGroupImpl groupImpl = (ReaderGroupImpl) getReaderGroup(groupName);
+
+        while (true) {
+            val state = groupImpl.getState();
+            switch (state.getConfigState()) {
+                case INITIALIZING:
+                    groupImpl.doInit(state);
+                    break;
+                case REINITIALIZING:
+                    groupImpl.doReinit(state);
+                    break;
+                case READY:
+                    AtomicBoolean b = new AtomicBoolean(true);
```

Comment:
rename this to something more meaningful for anyone reading the code. also please add a comment explaining in which cases could this boolean be false. And why "newstate" isnt sufficient. 

in fact this whole method signature appears very hacky. 
i know the intent is to know if our update succeeded. and we want to get the state after that and make use of it subsequently.
i think the method should return a boolean to indicate if the update succeeded. and the state can be retrieved from groupImpl.getState or something.

Reply:
i have added a comment on RGImpl

Reply:
I've changed the structure of this method to suit what is described below.